web/client/epics/context.js

```diff
@@ -133,3 +145,114 @@ export const handleLoginLogoutContextReload = action$ => {
+            const attribute = get(resource, 'Attributes.attribute', {});
+            const attributes = isArray(attribute) ? attribute : [attribute];
+            return get(find(attributes, ({name}) => name === 'thumbnail'), 'value');
+        };
+
+        return Observable.of(setControlProperty('mapTemplates', 'enabled', true, true)).concat(!mapTemplatesLoaded ?
+            Observable.defer(() => Api.searchListByAttributes(makeFilter(), {
+                params: {
+                    includeAttributes: true
+                }
+            }, '/resources/search/list'))
+                .switchMap((data) => {
+                    const resourceObj = get(data, 'ResourceList.Resource', []);
+                    const resources = isArray(resourceObj) ? resourceObj : [resourceObj];
+                    const newTemplates = resources.map(resource => ({
+                        ...pick(resource, 'id', 'name', 'description'),
+                        thumbnail: extractThumbnail(resource),
+                        dataLoaded: false,
+                        loading: false
+                    }));
+                    return Observable.of(setContext({...currentContext, templates: newTemplates}), setMapTemplatesLoaded(true));
+                })
+                .catch(err => Observable.of(setMapTemplatesLoaded(true, err))) :
+            Observable.empty()
+        );
+    });
+
+export const mergeTemplateEpic = (action$, store) => action$
+    .ofType(MERGE_TEMPLATE)
+    .switchMap(({id}) => {
+        const state = store.getState();
+        const templates = templatesSelector(state);
+        const template = find(templates, t => t.id === id);
+
+        return (template.dataLoaded ? Observable.of(template.data) :
+            Observable.defer(() => Api.getData(id))).switchMap(data => {
+            if (isObject(data) && data.map !== undefined) {
+                const map = mapSelector(state);
+                const layers = layersSelector(state);
+                const groups = groupsSelector(state);
+                const backgrounds = backgroundListSelector(state);
+                const textSearchConfig = textSearchConfigSelector(state);
+                const additionalOptions = mapOptionsToSaveSelector(state);
+
+                const currentConfig = MapUtils.saveMapConfiguration(map, layers, groups, backgrounds, textSearchConfig, additionalOptions);
+                const newConfig = MapUtils.mergeMapConfigs(currentConfig, data);
+
+                return Observable.of(...(!template.dataLoaded ? [setTemplateData(id, data)] : []), configureMap(cloneDeep(newConfig), null));
+            }
+
+            return !template.dataLoaded ? Observable.of(setTemplateData(id, data)) : Observable.empty();
+        }).let(wrapStartStop(
+            setTemplateLoading(id, true),
+            setTemplateLoading(id, false),
+            e => {
+                const messageId = errorToMessageId('template', e.originalError, store.getState);
```

Comment:
During testings I've seen this e or e.originalError doesn't exist, this causes an error sometime. Please try my context (`#/context/nali-templates`) and test adding layers and/or replacing randomly.

I think the error is caused by other issues in mapstore (google backgrounds, tileMatrixSet) anyway also these kind of errors should be handled.